Mathlib/LinearAlgebra/QuadraticForm/QuadraticModuleCat.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.LinearAlgebra.QuadraticForm.IsometryEquiv
+import Mathlib.Algebra.Category.AlgebraCat.Basic
+
+/-!
+# The category of quadratic modules
+-/
+
+open CategoryTheory
+
+universe v u
+
+variable (R : Type u) [CommRing R]
+
+/-- The category of quadratic modules; modules with an associated quadratic form-/
+structure QuadraticModuleCat extends ModuleCat.{v} R where
+  /-- The quadratic form associated with the module. -/
+  form' : QuadraticForm R carrier
+
+variable {R}
+
+namespace QuadraticModuleCat
+
+open QuadraticForm
+
+instance : CoeSort (QuadraticModuleCat.{v} R) (Type v) :=
+  ⟨(·.carrier)⟩
+
+instance (V : QuadraticModuleCat.{v} R) : AddCommGroup V :=
+  V.isAddCommGroup
+
+instance (V : QuadraticModuleCat.{v} R) : Module R V :=
+  V.isModule
+
+@[inherit_doc form']
+def form (V : QuadraticModuleCat.{v} R) : QuadraticForm R V := V.form'
+
+/-- See Note [custom simps projection] -/
+def Simps.form (V : QuadraticModuleCat.{v} R) : QuadraticForm R V := V.form
+
+initialize_simps_projections QuadraticModuleCat (form' → form)
+/-- The object in the category of quadratic R-modules associated to a quadratic R-module. -/
+@[simps form]
+def of {X : Type v} [AddCommGroup X] [Module R X] (Q : QuadraticForm R X) :
+    QuadraticModuleCat R where
+  form' := Q
+
+/-- A type alias for `QuadraticForm.LinearIsometry` to avoid confusion between the categorical and
+algebraic spellings of composition. -/
+@[ext]
```

Comment:
Once the `Category` instance has been defined, you should add the following lemma, otherwise the `ext` tactic will not find the automatically generated `Hom.ext`:
```lean
@[ext]
lemma hom_ext {M N : QuadraticModuleCat.{v} R} (f g : M ⟶ N) (h : f.toIsometry = g.toIsometry) :
    f = g :=
  Hom.ext _ _ h
```

Reply:
Yes, we have the exact same problem everytime we define a category...